reark/src/main/java/io/reark/reark/data/stores/cores/ContentProviderStoreCoreBase.java

```diff
@@ -180,44 +227,56 @@ public static Handler createHandler(@NonNull final String name) {
                             cursor.close();
                         }
 
-                        Log.v(TAG, "Create insertion operation for " + uri);
-                        return ContentProviderOperation.newInsert(uri)
-                                .withValues(getContentValuesForItem(item))
+                        Log.v(TAG, "Create insertion contentOperation for " + uri);
+                        ContentProviderOperation contentOperation = ContentProviderOperation
+                                .newInsert(uri)
+                                .withValues(getContentValuesForItem(value.item()))
                                 .build();
+
+                        return value.toOperation(contentOperation);
                     }
 
                     final U currentItem = read(cursor);
-                    final U newItem = mergedItem(currentItem, item);
+                    final U newItem = mergedItem(currentItem, value.item());
 
                     cursor.close();
 
                     if (!newItem.equals(currentItem)) {
-                        Log.v(TAG, "Create update operation for " + uri);
-                        return ContentProviderOperation.newUpdate(uri)
+                        Log.v(TAG, "Create update contentOperation for " + uri);
+                        ContentProviderOperation contentOperation = ContentProviderOperation
+                                .newUpdate(uri)
                                 .withValues(getContentValuesForItem(newItem))
                                 .build();
+
+                        return value.toOperation(contentOperation);
                     }
 
                     Log.v(TAG, "Data already up to date at " + uri);
-                    return NO_OPERATION;
+                    return value.noOperation();
                 })
-                .onErrorReturn(e -> NO_OPERATION)
-                .doOnNext(operation -> releaseIfNoOp(operation, uri))
-                .filter(ContentProviderStoreCoreBase::isValidOperation);
+                .onErrorReturn(__ -> value.noOperation())
```

Comment:
Aren't those 3 lines the same as the ones around 210 line? Can you somehow remove the duplication